modules/imaging_uploader/php/imaging_uploader.class.inc

```diff
@@ -299,34 +295,69 @@ class Imaging_Uploader extends \NDB_Menu_Filter_Form
+                );
             }
 
             ///////////////////////////////////////////////////////////////////////
             //////check to see if candid/pscid and visit-label exist///////////////
+            //////        check to see if candidate is active        //////////////
+            //////        check to see if session is active          //////////////
             ///////////////////////////////////////////////////////////////////////
             if ((!empty($candid)) && (!empty($pscid)) && (!empty($visit_label))) {
-                $result = $db->pselectOne(
-                    "SELECT COUNT(*) FROM session s
-                     JOIN candidate c ON (c.candid=s.candid)
-                     WHERE s.Visit_label = :vlabel
-                     AND c.pscid = :pid AND c.candid = :cid",
+                $candidate = $db->pselectRow(
+                    "SELECT c2.candid AS candid_with_pscid, s.visit_label,
+                     c2.active AS candidate_active,
+                     s.active AS session_active
+                     FROM candidate c
+                     LEFT JOIN candidate c2
+                     ON (c.candid=c2.candid AND c2.pscid = :pid)
+                     LEFT JOIN session s
+                     ON (s.candid=c.candid and s.visit_label = :vlabel)
+                     WHERE c.candid = :cid",
                     [
                      ':cid'    => $candid,
-                     ':vlabel' => $visit_label,
                      ':pid'    => $pscid,
+                     ':vlabel' => $visit_label,
                     ]
                 );
 
-                if ($result == 0) {
-                    $errors[] = "Make sure you enter the CandID: $candid,".
-                                " PSCID: $pscid and the corresponding".
-                                " Visit-label: $visit_label. Candidate and".
-                                " corresponding visit label must already be" .
-                                " created in the database. Please refresh this".
-                                " page to return to the Imaging Uploader module";
+                if (sizeOf($candidate) == 0) {
+                    array_push(
+                        $errors,
+                        "The candidate associated with that CandID".
+                        " does not exist in the database."
+                    );
+                } else {
+                    if (!$candidate['candid_with_pscid']) {
+                        array_push(
+                            $errors,
+                            "The CandID and the PSCID do not match."
+                        );
+                    }
+                    if (!$candidate['visit_label']) {
+                        array_push(
+                            $errors,
+                            "The candidate associated with that CandID".
+                            " and PSCID does not have that Visit Label."
```

Comment:
this comment is mostly a reminder. only valid visit labels should be given as options in the form anyways. this should be addressed in another PR. @victoriafoing has a ticket already been made for it?